lib/ansible/modules/network/cnos/cnos_vlan.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2017 Lenovo, Inc.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Module to send VLAN commands to Lenovo Switches
+# Overloading aspect of vlan creation in a range is pending
+# Lenovo Networking
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: cnos_vlan
+short_description: Performs VLAN switch configuration and state management.
+description:
+    - This module allows you to work with VLAN related configurations. The
+     operators used are overloaded to ensure control over switch VLAN
+     configurations. The first level of VLAN configuration allows to set up the
+     VLAN range, the VLAN tag persistence, a VLAN access map and access map
+     filter. After passing this level, there are five VLAN arguments that will
+     perform further configurations. They are vlanArg1, vlanArg2, vlanArg3,
+     vlanArg4, and vlanArg5. The value of vlanArg1 will determine the way
+     following arguments will be evaluated. For more details on how to use these
+     arguments, see [Overloaded Variables].
+     This module uses SSH to manage network device configuration.
+     The results of the operation can be viewed in results directory.
+     To know more about this module from Lenovo and customizing them for your
+     use cases, please visit our [User Guide](http://systemx.lenovofiles.com/
+     help/index.jsp?topic=%2Fcom.lenovo.switchmgt.ansible.doc%2Fansible_for_
+     cnos.html)
+version_added: "2.3"
+Options:
```

Comment:
No options specified, which is why Shippable is failing

docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html